optaplanner-core/src/test/java/org/optaplanner/core/impl/util/TestRandom.java

```diff
@@ -0,0 +1,223 @@
+package org.optaplanner.core.impl.util;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Random;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+
+import org.junit.jupiter.api.Assertions;
+
+/**
+ * On the later JDKs, it is no longer possible to mock {@link Random} to return custom sequences.
+ * Therefore we introduce this class to allow for that use case.
+ *
+ * It allows to provide a sequence of pre-defined "random" values.
+ * It maintains an internal count of values already returned and if larger than the sequence provided,
+ * the last element in the sequence is returned.
```

Comment:
Shouldn't it fail fast if it runs out?

Reply:
If a test uses more random values than were provided in the JUnit class, we probably want that to fail fast. Let's not take any chances?